backend/src/game/game.module.ts

```diff
@@ -3,9 +3,26 @@ import { TypeOrmModule } from '@nestjs/typeorm';
 import { Match } from 'src/entities/match.entity';
 import { MatchService } from 'src/match/match.service';
 import { GameGateway } from './game.gateway';
+import {
+  Friendship,
+  UserMatchHistory,
+  PendingFriendship,
+  User,
+  UserAvatars,
+} from '../entities/users.entity';
+import { UsersService } from '../users/users.service';
 
 @Module({
-  imports: [TypeOrmModule.forFeature([Match])],
-  providers: [GameGateway, MatchService],
+  imports: [
+    TypeOrmModule.forFeature([
+      User,
+      Friendship,
+      PendingFriendship,
+      UserAvatars,
+      Match,
+      UserMatchHistory,
+    ]),
+  ],
```

Comment:
このImportsが気になりました。

GameModuleの中ではこれらのテーブルは直接使ってない？と思うので、利用するモジュールをインポートすればいいのかも？と思いました。(ちょっと自信ないので、間違ってるかもしれないです；。。。)

```suggestion
  imports: [MatchModule, UsersModule],
```


これをするためには、User.module.tsとmatch.module.tsにexportsを設定する必要があります。
```
@Module({
  imports: [TypeOrmModule.forFeature([Match])],
  controllers: [MatchController],
  providers: [MatchService],
  exports: [MatchService],
})
export class MatchModule {}
```

Reply:
コメントありがとうございます！
なぜかdependencyエラーが出たままだったのですが、gameModuleの中プロバイダはUserService/MatchServiceを呼んでいたのが問題だったようです。が今はすべてうまくいっていると思います